docs/optimization-framework/ab-testing.md

```diff
@@ -0,0 +1,311 @@
+
+Altis provides an A/B Test Block, which you can use to test different variations of a block or set of blocks in your content, widgets and layouts. There is also a developer friendly framework to add new custom tests that is used to power built in features like Title and Featured Image tests which can be controlled from the Block Editor sidebar.
+
+Tests run until the end date or when a statistically significant improvement has been found, when a winner is found the winning variant will be shown to everyone.
+
+## Post titles and featured images A/B Tests
+
+With this feature enabled it's simple to create A/B Tests for your post titles and/or featured images directly from the post edit screen.
+
+It is enabled by default but can be disabled via the configuration file:
+
+```json
+{
+	"extra": {
+		"altis": {
+			"modules": {
+				"analytics": {
+					"native": {
+						"experiments": {
+							"titles": false,
+							"featured_images": false
+						}
+					}
+				}
+			}
+		}
+	}
+}
+```
+
+Within the post edit screen click on the A/B icon to access the Experiments panel. Here you can set up the variants for your tests, the amount of traffic to show the tests to as well as start and end dates.
+
+Once you are ready to run the test click on the toggle to unpause it. Results are updated every hour until a statistically significant winner is found.
+
+![A/B Testing Titles user interface](../assets/ab-tests-titles.png)
+
+By default post title and featured image A/B tests are enabled for Posts and Pages however custom post types can be supported using keys shown in the example below, either when registering the post type or on the `init` action, for example:
+
+```php
+add_action( 'init', function () {
+	// Add support for title A/B tests to the 'events' post type.
+	add_post_type_support( 'events', 'altis.experiments.titles' );
+
+	// Add support for featured image A/B tests to the 'events' post type.
+	add_post_type_support( 'events', 'altis.experiments.featured_images' );
+} );
+```
+
+Note that in order for the test to function correctly, it needs WordPress to output the `<ab-test></ab-test>` Web Component (generated by the `output_ab_test_html_for_post()` function) that includes all the variations of the feature representation, so the front-end would be able to switch to the proper variation to serve to the visitor. As such, the built-in features will only work if the native WordPress functions are used, which are `the_title()` or `get_the_title()` for post title and `the_post_thumbnail()` or `get_the_post_thumbnail()` for featured image.
+
+## Creating Custom Tests
+
+There is a programmatic API to register A/B tests for posts:
+
+**`Altis\Analytics\Experiments\register_post_ab_test( string $test_id, array $options )`**
+
+Sets up the test.
+
+- `$test_id`: A unique ID for the test.
+- `$options`: Configuration options for the test.
+  - `label <string>`: A human readable label for the test.
```

Comment:
Did we add another item here? `singular_label`? Should probably add those to example further down too

Reply:
Yeah, I even missed adding `label` to the examples, will do once we rewrite them as asked. Added to the function doc for now.